### PR TITLE
Oracle v2 dev

### PR DESCRIPTION
**Pull request by nms-7**
see #228

### CODE REVIEW

**Review by ahnaguib — APPROVED** [review]

File: contracts/MedianOracle.sol [hunk 1]
```diff
@@ -108,13 +108,21 @@ contract MedianOracle is OwnableUpgradeable, IOracle {
         Report[2] storage reports = providerReports[providerAddress];
         uint256[2] memory timestamps = [reports[0].timestamp, reports[1].timestamp];
 
+        // Checks that this providerAddress is already whitelisted
         require(timestamps[0] > 0);
 
         uint8 index_recent = timestamps[0] >= timestamps[1] ? 0 : 1;
         uint8 index_past = 1 - index_recent;
 
+        uint256 minValidTimestamp = block.timestamp.sub(reportExpirationTimeSec);
```
Comment (nms-7) [inline contracts/MedianOracle.sol:117]:
```suggestion
        uint256 minValidTimestamp = block.timestamp - reportExpirationTimeSec;
```

File: contracts/MedianOracle.sol [hunk 2]
```diff
@@ -108,13 +108,21 @@ contract MedianOracle is OwnableUpgradeable, IOracle {
...
+        uint256 maxValidTimestamp = block.timestamp.sub(reportDelaySec);
```
Comment (nms-7) [inline contracts/MedianOracle.sol:118]:
```suggestion
        uint256 maxValidTimestamp = block.timestamp - reportDelaySec;
```

**Comment by nms-7** [comment]
Updated `now` logic to `block.timestamp`. This is tested and ready for merge, but I'll hold til review @aalavandhan @brandoniles @ahnaguib

**Review by aalavandhan — APPROVED** [review]

**Review by aalavandhan — APPROVED** [review]

**Comment by aalavandhan** [comment]
Ref image 
image

File: contracts/MedianOracle.sol [hunk 3]
```diff
@@ -108,13 +108,21 @@ contract MedianOracle is OwnableUpgradeable, IOracle {
...
         uint8 index_past = 1 - index_recent;
 
+        uint256 minValidTimestamp = block.timestamp - reportExpirationTimeSec;
+        uint256 maxValidTimestamp = block.timestamp - reportDelaySec;
+
         // Check that the push is not too soon after the last one.
-        require(timestamps[index_recent] + reportDelaySec <= block.timestamp);
+        // unless past one is already expired
+        require(
+            timestamps[index_past] < minValidTimestamp ||
```
[the remark this answers is not in the document]
Comment (brandoniles) [reply contracts/MedianOracle.sol:123]: The boundaries are considered valid in the getData function, and it's mirrored here 👍

**Review by brandoniles — APPROVED** [review]
lgtm

**Pull request merged**